src/analysis/processing/qgsalgorithmexportmesh.cpp

```diff
@@ -668,4 +694,1021 @@ QSet<int> QgsExportMeshOnGrid::supportedDataType()
     QgsMeshDatasetGroupMetadata::DataOnVolumes} );
 }
 
+QString QgsMeshRasterizeAlgorithm::name() const
+{
+  return QStringLiteral( "meshrasterize" );
+}
+
+QString QgsMeshRasterizeAlgorithm::displayName() const
+{
+  return QObject::tr( "Rasterize mesh dataset" );
+}
+
+QString QgsMeshRasterizeAlgorithm::group() const
+{
+  return QObject::tr( "Mesh" );
+}
+
+QString QgsMeshRasterizeAlgorithm::groupId() const
+{
+  return QStringLiteral( "mesh" );
+}
+
+QString QgsMeshRasterizeAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Create a raster layer from a mesh dataset.\n"
+                      "For data on volume (3D stacked dataset values), the exported dataset values are averaged on faces using the method defined in the mesh layer properties (default is Multi level averaging method).\n"
```

Comment:
I am thinking to NOT allow 3d values in these algorithms at all. And create one extra algorithm "average 3d results", where you would select averaging method and persist it to the new dataset. that way user will be able to first average it out and then export ...